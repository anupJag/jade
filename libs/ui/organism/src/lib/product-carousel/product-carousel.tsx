import React from 'react';
import { Heading } from '@jade/ui/atoms';
import { Carousel } from '@jade/ui/molecules';
import { ProductCarouselTile } from '@jade/ui/organism';
import { Product } from '@jade/ui/organism/types';

type Props = {
  heading: string | any;
  products: Product[];
};

export const ProductCarousel = ({ heading, products }: Props) => {
  const Children = products.map((product: Product) => (
    <ProductCarouselTile key={product.sku} product={product} />
  ));

  if (Children.length > 0) {
    return (
      <div>
        <Heading tagName="h2">{heading}</Heading>
        <Carousel slidesToShow={3}>{Children}</Carousel>
      </div>
    );
  }

  return null;
};
