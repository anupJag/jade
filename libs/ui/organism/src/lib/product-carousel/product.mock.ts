import {
  Product,
  ProductUoM,
  ProductPricingSpecial,
  ProductPricing,
} from '@jade/ui/organism/types';

const mockProductUoM: ProductUoM = {
  unitType: 'kg',
  minQty: 1,
  price: 1.99,
  displayText: '1.99$',
};
const mockSpecialPricing: ProductPricingSpecial = {
  price: 1.68,
  from: '1.99$',
  to: '1.68$',
  amountSavings: 0.31,
  percentageSavings: 15,
};
const mockPricing: ProductPricing = {
  list: 1.68,
  retail: 1.99,
  special: [mockSpecialPricing],
};
const mockProduct: Product = {
  id: '1234',
  sku: '23970',
  title: 'sample image',
  shortDescription: 'sample product',
  description: ['sample product'],
  images: [{ url: './sample/image', altText: 'sample image', isDefault: false }],
  pricing: mockPricing,
  unitOfMeasure: mockProductUoM,
  productUrl: '/p/101',
  rating: { value: 4, count: 100 },
};
const products = [mockProduct];

export default products;
