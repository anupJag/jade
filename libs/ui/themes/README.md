# fe-themes

This library was generated with [Nx](https://nx.dev).

## Running unit tests

Run `ng test fe-themes` to execute the unit tests via [Jest](https://jestjs.io).
