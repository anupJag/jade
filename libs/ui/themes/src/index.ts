export * from './lib/defaultTheme';
export * from './lib/styled';
export * from './lib/themeProvider';
export { default as styled } from './lib/styled';
export * from './lib/GlobalStyles';
