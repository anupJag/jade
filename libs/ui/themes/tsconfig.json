{
  "extends": "../../../tsconfig.json",
  "compilerOptions": {
    "jsx": "react",
    "allowJs": false,
    "esModuleInterop": true,
    "allowSyntheticDefaultImports": true,
    "types": ["node", "jest"]
  },
  "include": ["**/*.js", "**/*.jsx", "**/*.ts", "**/*.tsx"]
}
