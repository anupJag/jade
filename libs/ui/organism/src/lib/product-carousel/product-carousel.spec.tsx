import React from 'react';
import { render } from '@testing-library/react';
import { ThemeProvider } from '@jade/ui/themes';
import mockProducts from './product.mock';

import { ProductCarousel } from './product-carousel';

describe('<ProductCarouselTile />', () => {
  let ProductCarouselComponent = null;
  beforeEach(() => {
    ProductCarouselComponent = render(
      <ThemeProvider>
        <ProductCarousel products={mockProducts} heading="test heading" />
      </ThemeProvider>,
    );
  });

  test('should render correctly', () => {
    const { asFragment } = ProductCarouselComponent;
    expect(asFragment()).toMatchSnapshot();
  });
});
