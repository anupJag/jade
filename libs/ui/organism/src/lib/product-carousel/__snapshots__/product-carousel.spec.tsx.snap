// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductCarouselTile /> should render correctly 1`] = `
<DocumentFragment>
  <div>
    <h2
      class="css-zhbmh3"
    >
      test heading
    </h2>
    <section
      class="css-13ug728"
    >
      <button
        aria-label="prev"
        class="css-q54x71"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="css-1im46kq"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
            fill="currentColor"
          />
        </svg>
      </button>
      <ul
        class="css-d1soyr"
      >
        <li
          class="css-ltvgdk"
          tabindex="0"
        >
          <div
            class="css-psphtj"
          >
            <div
              class="css-84pv12"
            >
              <img
                alt="sample image"
                class="css-82vs3h"
                height="150px"
                width="150px"
              />
            </div>
            <div
              class="css-0"
            >
              <a
                href="/p/101"
                tabindex="0"
              >
                <h3
                  class="css-zj698v"
                >
                  sample image
                </h3>
              </a>
              <div
                class="css-1upilqn"
              >
                ($1.68 / kg )
              </div>
              <div
                class="css-14w6evh"
              >
                $1.68
              </div>
              <button
                class="css-bbsppe"
                tabindex="0"
                type="button"
              >
                Add to Cart
              </button>
            </div>
          </div>
        </li>
      </ul>
      <button
        aria-label="next"
        class="css-13t3hkw"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="css-1im46kq"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
            fill="currentColor"
          />
        </svg>
      </button>
      <div
        class="css-3buig3"
      >
        <ul
          class="css-k008qs"
        >
          <li
            class="css-i6zvb9"
          >
            <button
              aria-label="item 1 of 1"
              class="css-1uqak3w"
            >
               
            </button>
          </li>
        </ul>
      </div>
    </section>
  </div>
</DocumentFragment>
`;
